datafile/openers/xdg.lua

```diff
@@ -13,9 +13,9 @@ local function split(var)
 end
 
 local HOME = os.getenv("HOME")
-local XDG_DATA_HOME   = os.getenv("XDG_DATA_HOME")   or HOME.."/.local/share"
-local XDG_CONFIG_HOME = os.getenv("XDG_CONFIG_HOME") or HOME.."/.config"
-local XDG_CACHE_HOME  = os.getenv("XDG_CACHE_HOME")  or HOME.."/.cache"
+local XDG_DATA_HOME   = os.getenv("XDG_DATA_HOME")   or (HOME and HOME.."/.local/share")
+local XDG_CONFIG_HOME = os.getenv("XDG_CONFIG_HOME") or (HOME and HOME.."/.config")
+local XDG_CACHE_HOME  = os.getenv("XDG_CACHE_HOME")  or (HOME and HOME.."/.cache")
```

Comment:
as a result of this it can happen that `XDG_CACHE_HOME == nil`.
In that case, then slightly further down we get an array with holes;
```lua
local XDG_CACHE_DIRS = { XDG_CACHE_HOME, XDG_RUNTIME_DIR }
```
So I think that line should also be changed to;
```lua
local XDG_CACHE_DIRS = { XDG_CACHE_HOME }
table.insert(XDG_CACHE_DIRS, XDG_RUNTIME_DIR)
```

Reply:
I've fixed it in a slightly different way, thanks!